allennlp/modules/time_distributed.py

```diff
@@ -15,28 +17,55 @@ class TimeDistributed(torch.nn.Module):
 
     Note that while the above gives shapes with ``batch_size`` first, this ``Module`` also works if
     ``batch_size`` is second - we always just combine the first two dimensions, then split them.
+
+    It also reshapes keyword arguments unless they are not tensors or their name is specified in
+    the optional ``pass_through`` iterable.
     """
     def __init__(self, module):
-        super(TimeDistributed, self).__init__()
+        super().__init__()
         self._module = module
 
-    def forward(self, *inputs):  # pylint: disable=arguments-differ
-        reshaped_inputs = []
-        for input_tensor in inputs:
-            input_size = input_tensor.size()
-            if len(input_size) <= 2:
-                raise RuntimeError("No dimension to distribute: " + str(input_size))
+    def forward(self, *inputs, pass_through: Optional[Iterable] = None,  # pylint: disable=arguments-differ
```

Comment:
Minor formatting thing: this would be easier to read if you put the `**kwargs):` on this line, and moved the pylint command to the line below (the first line of `forward`).

Reply:
Also, for the type annotation on `pass_through`, use `List[str]`, without any `Optional` - the `= None` means the same thing as `Optional`, so `Optional` is redundant.

Reply:
> Also, for the type annotation on `pass_through`, use `List[str]`, without any `Optional` - the `= None` means the same thing as `Optional`, so `Optional` is redundant.

Why `List` and not `Iterable`?

Reply:
Because that's what we use everywhere else in the code.  The type annotations are mostly for readability, and for simple mypy checks, not for strict compliance with types everywhere.  So, we don't need to be as generic as possible with our types, as you might be in some other languages.

Reply:
Changed it.

Actually, in this case, it should be `Container` and not `Iterable`, because `__contains__` is the only operation that matters here for `pass_through`.

I kinda disagree on this one though. First, I understand this is a separate issue, so it shouldn't be addressed here (and should be addressed for the codebase overall then, if so). Second, I believe there's a reason why such typing annotations exist (e.g., `Container` and `Iterable`). I use PyCharm, and if in a similar scenario I pass a `set` instead of a `list`, there's a chance it'll show a warning and I think it's okay because the method expects a `list`. But in a case like this, it should be fine to pass a `set`. On the other hand, it can mean to "complicate" stuff and also cause a maintenance burden because method signatures could potentially change all the time (e.g., because `pass_through` is used to be iterated after certain change) or maintainers forget to change it, albeit I'm not sure how frequently it could happen. Any thoughts on this?

Reply:
https://google.github.io/styleguide/pyguide.html#4-parting-words

Our type hints and our use of type checkers are purely an aid to our sanity, not the other way around.  We shouldn't make ourselves go into contortions just to make a type checker happy; that's counter productive.  The bottom line: we have a lot of code here that uses `List`, because that's the easiest intuitive type annotation for a human to read, even if a `Set` might also be appropriate, so we really should be using something more general.  Be consistent with the code around you.  For similar reasons, and because the `Optional` really doesn't add anything except for the type checker, just remove the `Optional` annotation.

Reply:
Oh, didn't know that rule. I like it and approve it.

What you say totally makes sense. I've already removed the `Optional` thing. Thanks for the feedback!

Reply:
One last point - there is a point where the difference between an annotation of `List` and `Iterable` makes a big difference.  That's in our [`FromParams` code](https://github.com/allenai/allennlp/blob/84eb4d1acc03bad89e2282306eb4ede8f1badec5/allennlp/common/from_params.py#L176-L212).  If we see a type annotation of `Iterable`, we don't know how to instantiate it.  I guess we could default to constructing a list, or something, but it's a lot more clear if you just give the type that you want.

Reply:
I see, interesting point. But does it represent a problem even though this module is not registered?

Reply:
No, not for this module, and especially not for the `forward` method.  But that's another reason why we just tend to use `List` annotations instead of the more abstract classes.